g2p/cli.py

```diff
@@ -183,7 +339,46 @@ def generate_mapping(in_lang, out_lang, dummy, ipa, list_dummy, out_dir, merge):
             new_mapping.config_to_file()
             new_mapping.mapping_to_file()
 
+    elif from_langs is not None:
+        # --from/--to mode
+        assert to_langs is not None
+
+        from_mappings = []
+        for from_lang in from_langs.split(":"):
+            from_mappings.extend(parse_from_or_to_lang_spec(from_lang))
+        to_mappings = []
+        for to_lang in to_langs.split(":"):
+            to_mappings.extend(parse_from_or_to_lang_spec(to_lang))
+
+        if not from_mappings:
+            raise click.UsageError(
+                f'Invalid --from value "{from_langs}": no mappings found.'
+            )
+        if not to_mappings:
+            raise click.UsageError(
+                f'Invalid --to value "{to_langs}": no mappings found.'
+            )
+
+        for from_mapping, in_or_out in from_mappings:
+            LOGGER.info(
+                f'From mapping: {from_mapping.kwargs["in_lang"]}_to_{from_mapping.kwargs["out_lang"]},{in_or_out}'
+            )
+        for to_mapping, in_or_out in to_mappings:
+            LOGGER.info(
+                f'To mapping: {to_mapping.kwargs["in_lang"]}_to_{to_mapping.kwargs["out_lang"]},{in_or_out}'
+            )
+
+        new_mapping = create_multi_mapping(from_mappings, to_mappings)
+
+        if out_dir:
+            new_mapping.config_to_file(out_dir)
+            new_mapping.mapping_to_file(out_dir)
+        else:
+            new_mapping.config_to_file()
+            new_mapping.mapping_to_file()
+
 
+# TODO the path argument is ignored. What was it supposed to do? Code it...
```

Comment:
Was it an output path for the generated mapping?

Reply:
You weren't paying enough attention reviewing this one... :) That's for `g2p generate_mapping_network`, code you wrote back in 2019, and I guess you meant to save the output there (it's a required directory name that has to already exist), but never got around to it.